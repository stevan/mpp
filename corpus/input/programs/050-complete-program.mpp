package Calculator;
use strict;

sub add($x, $y) {
    return $x + $y;
}

sub multiply($x, $y) {
    return $x * $y;
}

my $a = 5;
my $b = 10;
my $sum = add($a, $b);
my $product = multiply($a, $b);

print($sum);
print($product);