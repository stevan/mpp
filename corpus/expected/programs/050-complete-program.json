[
  {
    "type": "Package",
    "name": "Calculator"
  },
  {
    "type": "Use",
    "module": "strict"
  },
  {
    "type": "Sub",
    "name": "add",
    "parameters": [
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$x"
        }
      },
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$y"
        }
      }
    ],
    "body": [
      {
        "type": "Return",
        "value": {
          "type": "BinaryOp",
          "operator": "+",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Variable",
            "name": "$y"
          }
        }
      }
    ]
  },
  {
    "type": "Sub",
    "name": "multiply",
    "parameters": [
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$x"
        }
      },
      {
        "type": "Parameter",
        "variable": {
          "type": "Variable",
          "name": "$y"
        }
      }
    ],
    "body": [
      {
        "type": "Return",
        "value": {
          "type": "BinaryOp",
          "operator": "*",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Variable",
            "name": "$y"
          }
        }
      }
    ]
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$a"
    },
    "initializer": {
      "type": "Number",
      "value": "5"
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$b"
    },
    "initializer": {
      "type": "Number",
      "value": "10"
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$sum"
    },
    "initializer": {
      "type": "Call",
      "name": "add",
      "arguments": [
        {
          "type": "Variable",
          "name": "$a"
        },
        {
          "type": "Variable",
          "name": "$b"
        }
      ]
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$product"
    },
    "initializer": {
      "type": "Call",
      "name": "multiply",
      "arguments": [
        {
          "type": "Variable",
          "name": "$a"
        },
        {
          "type": "Variable",
          "name": "$b"
        }
      ]
    }
  },
  {
    "type": "Call",
    "name": "print",
    "arguments": [
      {
        "type": "Variable",
        "name": "$sum"
      }
    ]
  },
  {
    "type": "Call",
    "name": "print",
    "arguments": [
      {
        "type": "Variable",
        "name": "$product"
      }
    ]
  }
]